bundles/net.tourbook/src/net/tourbook/ui/views/TourNutritionView.java

```diff
@@ -0,0 +1,1628 @@
+
+      /*
+       * Add product button
+       */
+      final Button btnAddCustomProduct = new Button(parent, SWT.NONE);
+      btnAddCustomProduct.setText(Messages.Tour_Nutrition_Button_AddCustomProduct);
+      btnAddCustomProduct.setToolTipText(Messages.Tour_Nutrition_Button_AddCustomProduct_Tooltip);
+      btnAddCustomProduct.addSelectionListener(widgetSelectedAdapter(selectionEvent -> {
+
+         final DialogCustomTourNutritionProduct dialogTourNutritionProduct = new DialogCustomTourNutritionProduct(Display.getCurrent()
+               .getActiveShell());
+
+         if (dialogTourNutritionProduct.open() != Window.OK) {
+            return;
+         }
+
+         final Set<TourNutritionProduct> tourNutritionProducts = _tourData.getTourNutritionProducts();
+         tourNutritionProducts.add(dialogTourNutritionProduct.getTourNutritionProduct(_tourData));
+         _tourData.setTourNutritionProducts(tourNutritionProducts);
+         _tourData = TourManager.saveModifiedTour(_tourData);
+         _tourData.setTourNutritionProducts(_tourData.getTourNutritionProducts());
+
+      }));
+      btnAddCustomProduct.setImage(_imageAdd);
+      GridDataFactory.fillDefaults().align(SWT.BEGINNING, SWT.CENTER).grab(true, false).applyTo(btnAddCustomProduct);
+
+      /*
+       * Delete product button
+       */
+      _btnDeleteProduct = new Button(parent, SWT.NONE);
+      _btnDeleteProduct.setText(Messages.Tour_Nutrition_Button_DeleteProduct);
+      _btnDeleteProduct.setToolTipText(Messages.Tour_Nutrition_Button_DeleteProduct_Tooltip);
+      _btnDeleteProduct.setEnabled(false);
+      _btnDeleteProduct.addSelectionListener(widgetSelectedAdapter(selectionEvent -> onDeleteProducts()));
+      _btnDeleteProduct.setImage(_imageDelete);
+      GridDataFactory.fillDefaults().align(SWT.END, SWT.CENTER).grab(true, false).applyTo(_btnDeleteProduct);
+   }
+
+   private void createUI_220_Viewer(final Composite parent) {
+
+      /*
+       * table viewer: products
+       */
+      final Table productsTable = new Table(parent, SWT.MULTI | SWT.FULL_SELECTION);
+      GridDataFactory.fillDefaults().grab(true, true).span(3, 1).applyTo(productsTable);
+      productsTable.setLinesVisible(_prefStore.getBoolean(ITourbookPreferences.VIEW_LAYOUT_DISPLAY_LINES));
+      productsTable.setHeaderVisible(true);
+
+      final Listener paintListener = event -> {
+
+         // paint images at the correct column
+         final int columnIndex = event.index;
+         if (columnIndex == _columnIndex_ForColumn_IsBeverage &&
+               event.type == SWT.PaintItem) {
+
+            onPaint_Viewer_GraphImage(event);
+         }
+      };
+      productsTable.addListener(SWT.MeasureItem, paintListener);
+      productsTable.addListener(SWT.PaintItem, paintListener);
+
+      productsTable.addControlListener(controlResizedAdapter(controlEvent -> setWidth_ForColumn_IsBeverage()));
+
+      _productsViewer = new TableViewer(productsTable);
```

Comment:
There should be a viewer context menu with the actions

* Open product website (the double click action is an invisible feature)
* Edit
* Delete (the delete button in the action bar may be could be removed)

Reply:
Contextual menu added

I kept the "Edit" button for[ a future improvement](https://github.com/mytourbook/mytourbook/issues/1280#possible-future-improvements)